test/common/common/utility_test.cc

```diff
@@ -116,6 +118,65 @@ TEST(DateUtil, NowToMilliseconds) {
   EXPECT_EQ(12345067, DateUtil::nowToMilliseconds(test_time));
 }
 
+TEST(OutputBufferStream, FailsOnWriteToEmptyBuffer) {
+  constexpr char data = 'x';
+  OutputBufferStream ostream{nullptr, 0};
+  ASSERT_TRUE(ostream.good());
+
+  ostream << data;
+
+  EXPECT_TRUE(ostream.bad());
+}
+
+TEST(OutputBufferStream, CanWriteToBuffer) {
+  constexpr char data[] = "123";
+  std::array<char, 3> buffer;
```

Comment:
I forget if ASAN is able to find issues with data members like this allocated in the stack.  Consider instead:

std::unique_ptr<char[]> buffer(new char[3]);

Reply:
If you were to do something like:

OutputBufferStream ostream{buffer.data(), buffer.size()+1};
ostream << "lots of bytes";

Would ASAN detect the problem?

Reply:
Yes ASAN detected that. 

We're saying the buffer is bigger than it is (i.e. buffer.size() + 1) and trying to write to memory we don't have in that case.

Reply:
Great.  std::array works for me.